client-library/src/Attestation/AttestationClient/lib/AttestationClientImpl.cpp

```diff
@@ -172,6 +177,48 @@ AttestationResult AttestationClientImpl::Attest(const ClientParameters& client_p
     return result;
 }
 
+AttestationResult AttestationClientImpl::GetHardwarePlatformEvidence(std::string &evidence,
+                                                                     const std::string &client_payload) noexcept {
+    AttestationResult result(AttestationResult::ErrorCode::SUCCESS);
+
+    CLIENT_LOG_INFO("Getting Td Report from driver...");
+
+    std::unique_ptr<std::string> buffer = std::make_unique<std::string>(TD_REPORT_SIZE, 0);
+    int status = TDX_GET_REPORT_SUCCESS;
+    if (!client_payload.empty()) {
+        unsigned char hash[SHA256_DIGEST_LENGTH];
+        SHA256((unsigned char *)client_payload.data(), client_payload.length(), hash);
+        status = GetTdReport((char *)buffer.get()->data(), hash, SHA256_DIGEST_LENGTH);
+    }
+    else {
+        status = GetTdReport((char *)buffer.get()->data(), NULL, 0);
+    }
+
+    if (status == TDX_GET_REPORT_FAILED) {
+        CLIENT_LOG_ERROR("Failed to get report from tdx driver");
+        return AttestationResult::ErrorCode::ERROR_READING_TD_REPORT;
+    }
+    std::string hardware_report(buffer.release()->data(), TD_REPORT_SIZE);
```

Comment:
Can we just use buffer? Not sure why we need to create a new variable here.  Also release() releases ownership of the pointer, so we have to make sure we free that memory.

Reply:
Yes, next commit will address this!